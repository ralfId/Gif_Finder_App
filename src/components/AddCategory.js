import React, { useState } from "react";
import PropTypes from 'prop-types'


export const AddCategory = ({setCategory}) => {
  const [inputValue, setInputValue] = useState('');

  const handleOnChangeInput = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) =>{
      e.preventDefault();
      console.log('handleSubmit', inputValue);
      if ( inputValue.trim().length > 2 ) {
        setCategory(catg => [inputValue, ...catg] );
      }
      setInputValue('');
  };

  return (
    <form onSubmit={handleSubmit}>
        <p>{inputValue}</p>
        <input onChange={handleOnChangeInput} value={ inputValue } type="text" />

    </form>
  );
};

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}
