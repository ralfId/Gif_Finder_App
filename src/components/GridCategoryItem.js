import React from 'react'
import PropTypes from 'prop-types';

export const GridCategoryItem = ({ title, img }) => {
    return (
        <div className='card animate__fadeInLeft'>
            <img src={img} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GridCategoryItem.protoTypes = {
    title: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired
};