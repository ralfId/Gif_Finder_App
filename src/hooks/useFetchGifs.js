import { useEffect, useState } from "react"
import { GetGif } from "../helpers/getGifs";


export const useFetchGifs = (category) => {
  
    const [state, setState] = useState(
        {
            data: [],
            loading: true
        }
    );


    useEffect(()=>{
        GetGif(category)
        .then(imgs =>
            {
                setState({
                    data: imgs,
                    loading: false
                })
            }
            );
    });

    

    return state;
}
