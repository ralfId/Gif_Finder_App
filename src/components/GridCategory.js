import React, { useState, useEffect } from 'react'
import { GetGif } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GridCategoryItem } from './GridCategoryItem';

export const GridCategory = ({ category }) => {


    // const [gifImg, setGifImg] = useState([])
    const {data:gifImg, loading} = useFetchGifs(category);
    
    // useEffect(() => {
    //     GetGif(category).then(setGifImg);
    // }, [category]);

    

    return (
        <>
            <h3>{category}</h3>
            {loading && <p>Loading...</p>}
            <div className='card-container animate__zoomIn'>
                {
                    gifImg.map(img => (
                        <GridCategoryItem
                            key={img.id} {...img} />
                    ))
                }
            </div>


        </>
    )
}
