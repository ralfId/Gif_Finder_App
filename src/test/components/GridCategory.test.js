import { shallow } from "enzyme";
import { GridCategory } from "../../components/GridCategory";
import { useFetchGifs } from "../../hooks/useFetchGifs";
/** 
 * Finje cualquier llamada al archivo especificado en el path de jest.mock('path')
 * y supone o controla la info que esa llamada debe responder
 */
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en GridCategory.js', () => {
    
    const category = 'one piece';

    test('Debe hacer match con el snapshop ', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true 
        });

        const wraper = shallow(<GridCategory category={category}/>)
        expect(wraper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imagenes con useFetchGifs.js', () => {
        
        const gifData = [
            {
                id: 1,
                url: 'https://imagen.gif.com',
                title: 'Lorem ipsum'
            }
            
        ]
        useFetchGifs.mockReturnValue({
            data: gifData,
            loading: false 
        });

        const wraper = shallow(<GridCategory category={category}/>)
        expect(wraper).toMatchSnapshot();
        expect(wraper.find('p').exists()).toBe(false);
        expect(wraper.find('GridCategoryItem').length).toBe(gifData.length)
    });
});