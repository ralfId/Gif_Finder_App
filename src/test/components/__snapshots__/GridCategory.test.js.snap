// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GridCategory.js Debe de mostrar items cuando se cargan imagenes con useFetchGifs.js 1`] = `
<Fragment>
  <h3>
    one piece
  </h3>
  <div
    className="card-container animate__zoomIn"
  >
    <GridCategoryItem
      id={1}
      key="1"
      title="Lorem ipsum"
      url="https://imagen.gif.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en GridCategory.js Debe hacer match con el snapshop  1`] = `
<Fragment>
  <h3>
    one piece
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-container animate__zoomIn"
  />
</Fragment>
`;
